test/unit/org/apache/cassandra/schema/LegacySchemaMigratorThriftTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.schema;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import org.apache.cassandra.config.Schema;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import static org.apache.cassandra.cql3.QueryProcessor.executeOnceInternal;
+
+@SuppressWarnings("deprecation")
+public class LegacySchemaMigratorThriftTest extends LegacySchemaMigratorBaseTest
+{
+    public static final String KEYSPACE_18956 = "ks18956";
```

Comment:
make these two private

Reply:
The base class need to using table name to decide whether to use serializeKind or serializeKind18956. The bug needs `serializeKind18956` to reproduce.
```
        if (table.cfName.equals(LegacySchemaMigratorThriftTest.TABLE_18956)) {
            adder.add("validator", column.type.toString())
                    .add("type", serializeKind18956(column.kind))
                    .add("component_index", column.position());
        } else {
            adder.add("validator", column.type.toString())
                    .add("type", serializeKind(column.kind, table.isDense()))
                    .add("component_index", column.position());
        }
```
I can remove the static tag for the base methods and abstract `serializeKind` function so that the two child tests can implement its own `serializeKind` function. Do you think that makes sense?

Reply:
yes that makes sense.

Reply:
I have modified them to private and abstract the `serializeKind` function.